backend/poll.go

```diff
@@ -0,0 +1,184 @@
+		return answers, err
+	}
+	return answers, nil
+}
+
+func saveAnswerIntoDatastore(context context.Context, answerText string, voteCount int) error {
+	answer := &Answer{
+		AnswerText: answerText,
+		Votes:      voteCount,
+	}
+	answerKey := datastore.NewKey(context, "Answer", answerText, 0, nil)
+	_, error := datastore.Put(context, answerKey, answer)
+	if error != nil {
+		return error
+	}
+	return nil
+}
+
+func updateAnswerCount(answers []Answer, answerText string) int {
+	voteCount := 1
+	found := false
+	for i := 0; i < len(answers); i++ {
+		if answers[i].AnswerText == answerText {
+			voteCount = answers[i].Votes + 1
+			answers[i].Votes = voteCount
+			found = true
+			break
+		}
+	}
+	if !found {
+		answers = append(answers, Answer{
+			AnswerText: answerText,
+			Votes:      voteCount,
+		})
+	}
+	return voteCount
+}
+
+func calculatePollResults(w http.ResponseWriter, ctx context.Context, pollForm PollForm) ([]Result, error) {
+	var results []Result
+	err := datastore.RunInTransaction(ctx, func(ctx context.Context) error {
+		//create keys
+		clientIdKey := datastore.NewKey(ctx, "PollForm", pollForm.ClientId, 0, nil)
+		answerKey := datastore.NewKey(ctx, "Answer", pollForm.Answer, 0, nil)
+		//check if the user has already answered
+		var existingPollForm PollForm
+		err := datastore.Get(ctx, clientIdKey, &existingPollForm)
+		//error is nil if the user has answered
+		if err == nil {
+			//decrement the vote for the existing answer
+			var answer Answer
+			existingPollFormKey := datastore.NewKey(ctx, "Answer", existingPollForm.Answer, 0, nil)
+			err = datastore.Get(ctx, existingPollFormKey, &answer)
+			//decrement the vote for the answer
+			answer.Votes--
+			//update the existing answer
+			_, err = datastore.Put(ctx, existingPollFormKey, &answer)
+		}
+		//persist association between clientId and answer
+		_, err = datastore.Put(ctx, clientIdKey, &pollForm)
+		//get the lastest vote count for the answer
+		var answer Answer
+		err = datastore.Get(ctx, answerKey, &answer)
+		answer.AnswerText = pollForm.Answer
+		//increment the vote for the answer
+		answer.Votes++
+		//persist the answer
+		_, err = datastore.Put(ctx, answerKey, &answer)
+		return err
+	}, &datastore.TransactionOptions{XG: true})
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return results, err
+	}
+
+	answers, error := getAnswersFromDatastore(ctx)
+	if error != nil {
+		return results, error
+	}
+	for _, value := range answers {
+		results = append(results, Result{Answer: value.AnswerText, Percentage: make([]int, value.Votes)})
+	}
+	return results, nil
+}
+
+func parsePollForm(w http.ResponseWriter, r *http.Request, context context.Context) (PollForm, error) {
+	answerText, answerErr := parseNotEmptyFormValue(r, "answer")
+	clientId, clientIdErr := parseNotEmptyFormValue(r, "clientId")
+
+	pollForm := PollForm{clientId, answerText}
+
+	error := parseFormErrors([]error{answerErr, clientIdErr})
+	if error != nil {
+		return pollForm, error
```

Comment:
I can't use nill, I get the error:
"cannot use nil as type PollForm in return argument"